### ISSUE TITLE
Long delay after begin

### ISSUE THREAD

**Issue by metanav** [arduino/arduino-alvik-mpy#30]
Why is there long delay (around 12 seconds) when Alvik.begin() called? Looking at the code it seems doing version compatibility check only.  I used the code below.

```
import time
from arduino_alvik import ArduinoAlvik

start_time = time.time()
alvik = ArduinoAlvik()
alvik.begin()
end_time = time.time()

print(f'\nTotal time: {end_time - start_time}s')
print(f'\nlib_version: {alvik.get_lib_version()} fw_version: {alvik.get_fw_version()}')

```

Output:

```
Total time: 12s

lib_version: 1.1.0 fw_version: 1.1.0
```

**Comment by gbr1** [comment]
Hi @metanav !

Please can you tell me if in these 12seconds both lights are on?
Are you using stock firmware updated via [alvikupdate.arduino.cc](alvikupdate.arduino.cc) ?

Thank you in advance!

**Comment by gbr1** [comment]
Please @metanav  could you try this: https://alvikupdate.arduino.cc ?
Then turn off the robot and detach the usb cable.
See if it happens again

**Comment by metanav** [comment]
Now I am using the Arduino Alvik Updater as you suggested. By the way, how long does it take to complete? The status showing "Removing Folder" since 15 minutes.

![Image](https://github.com/user-attachments/assets/669f43ce-3c20-4d68-bbaa-e3a3a00acd7b)

**Comment by metanav** [comment]
MicroPython v1.24.0 on 2024-10-25; Arduino Nano ESP32 with ESP32S3

**Comment by gbr1** [comment]
Wait, you cannot use the updater if your board isn't running micropython. 
When you run an arduino sketch the flash is erased, so there isn't micropython. If you rub an Arduino sketch does it make 12s issue?

**Comment by metanav** [comment]
I uploaded the empty sketch again. When I connect it using the Arduino Lab for Micropython, It displays the REPL:

```
MicroPython v1.24.0 on 2024-10-25; Arduino Nano ESP32 with ESP32S3
Type "help()" for more information.
>>> 
```

It means Micropython did not remove.

**Comment by gbr1** [comment]
it could corrupted because part of the flash is overwritten

**Comment by metanav** [comment]
I erased the flash using the command:

```
esptool.py --chip esp32s3 --port /dev/tty.usbmodem142101   --baud 115200 erase_flash
```
Now Arduino Lab for Micropython cannot connect to it. So Micropython is gone. Let me know how to upload 1.22.

**Comment by gbr1** [comment]
Please check here:
https://micropython.org/download/ARDUINO_NANO_ESP32/
download with `1.22.2` and use [Micropython Installer](https://labs.arduino.cc/en/labs/micropython-installer)

**Comment by metanav** [comment]
I could revert it to 1.22.2 and uploaded the firmware using the alvikupdate.arduino.cc. Now it takes 6 seconds (seems consistent) to begin. Is it normal?

```
>>> 
MicroPython v1.22.2 on 2024-02-22; Arduino Nano ESP32 with ESP32S3
Type "help()" for more information.
>>> 
raw REPL; CTRL-B to exit
                  
Total time: 6s

lib_version: 1.1.0 fw_version: 1.1.0
```

**Comment by gbr1** [comment]
Yes! 6s is the normal time to boot.
It seems to be long but consider this:
```
1. begin() reset the stm32 on the alvik carrier
2. stm32 initialize sensors (tof require between 1.5s-2s to be ready)
3. wait for ack and version
4. check version
```

I'm trying to install micropython 1.24 to see what happens

**Comment by gbr1** [comment]
You are right, 1.24 introced something which makes alvik very slow

**Comment by ubidefeo** [comment]
@gbr1 
I have installed 1.24 on my Alvik and indeed it is very slow at starting up, but I cannot figure out where the hang-up is.
1.24 has definitely broken something 😔

**Comment by ubidefeo** [comment]
@metanav 

we are currently testing an internal build of MicroPython using IDF 5.2.3 and it seems to fix the UART issues.
an issue has been opened on MicroPython's GH (https://github.com/micropython/micropython/issues/16173)

Damien has attached the build to the thread if you care to test it.
Maybe 1.24.1 will be built against 5.2.3, but like @gbr1 said you're safer with 1.22.2 for now until we finalise all our testing

**Comment by eigen-value** [comment]
@gbr1 we may close this one as solved by #35

**Comment by gbr1** [comment]
hi @metanav we made a new release with a workaround to solve micropython 1.24 bugs, please check here: https://github.com/arduino/arduino-alvik-mpy/releases/tag/1.1.2
